### DISCUSSION TITLE
Does RaspAP have a firewall enabled?

### DISCUSSION THREAD

**Discussion by nextcloudfan** [RaspAP/raspap-webgui#1608] [Q&A] [answered]
Hi does RaspAP a firewall enabled? I understand, that it has no GUI for configuring a firewall. But are there any rules present?

Which ones?

Thanks & best

**Comment by billz** [comment]
Check the docs? There's an optional firewall https://docs.raspap.com/firewall/

**Comment by nextcloudfan** [comment]
I know this site. Again:
Does it mean devs deploy this project without the minimalistic security?! Are the basic rules enabled? Or are even this basic rules missing?

**Reply by billz** [reply]
RPi OS, and most Linux distros, do not include any kind of pre-configured firewall. If you wish to implement one, then install ufw from the apt repository. The optional firewall implementation from RaspAP is provided as a convenience. If your RaspAP device is behind a home router, you probably don't need one.

If you've spent any time in the official docs, you'd know about [the basic iptables rules](https://docs.raspap.com/manual/#routing-and-ip-masquerading) used by this project (try searching for 'firewall').

**Comment by SirLagz** [comment, marked as answer]
Most deployments of Rasp AP will be behind existing firewalls. If you're deploying a Rasp AP instance that's accessible from the Internet, then there are general linux hardening guides that should be followed.
